Automattic/sensei
# Show notice in Course Editor to focus on Course Outline block or Insert one if block does not exist for first time users

Resolves https://github.com/Automattic/sensei/issues/7331

## Proposed Changes
New users often don't know where they can add lessons, or sometimes, they don't know about the need of Course Outline block. So here, we've added a notice in the Course Editor that is shown to only those users who have not created any course yet.

Clicking on the link on that notice will focus (and scroll to, if necessary) on the Course Outline block if there is already one added in the editor.

If the Course Outline block is not there in the editor, it'll create one for the user. If at least one of the lessons in the Course Outline block is published, this notice will disappear.

To not bother the existing users, we've taken some measures. If the user has authored any course, this notice isn't shown to them. Once the user dismisses the notice, it's not shown to them anymore.

One thing to be noted is, that the dismissed flag is stored in localStorage. So the user may see it in a different browser or device. But it may not be very likely because once the user is in the Course editor, and playing around with it, there's a high probability that they'll have authored at least a course. So I didn't want to add another user meta for it as it may turn out to be not very useful.

I'll add some more tests for testing all the edge cases either in this PR, or a new one. I'm temporarily moving on to another issue.

## Testing Instructions


1. Create a new User who can create course
2. Create a new Course
3. Check that you see the desired notice on top
4. Click on the offline outline (Start with blank) or generate lessons with AI (Generate with AI)
5. Click on the `add some lessons` link on the notice
6. Make sure it takes the focus to the Course Outline block
7. Now delete all content from the editor
8. Now insert a pattern from Block Inserter (+ button on Top Toolbar of GB) -> Patterns -> Sensei LMS -> (Select any long pattern)
9. Now click on the 'add some lessons' link in the notice again
10. Make sure it scrolls you to the Course Outline block
11. Now remove everything again, or at least the Course Outline block
12. Click on the aforementioned link
13. Make sure it adds a Course Outline block in the editor
14. Now click on a lesson, from its toolbar, click 'Edit Lesson', and in the lesson editor, publish the lesson. Remember you are publishing only the lesson, not the course
15. Go back to the Course editor
16. Make sure the notice isn't there anymore
17. Create another similar user
18. Create a course. This time in the editor, dismiss the notice
19. Refresh the window, make sure the notice isn't appearing anymore
20. Now log in with an user who has authored at least one course
21. Make sure you don't see the notice for the user

https://github.com/Automattic/sensei/assets/6820724/cf3c8b30-0306-4229-930f-dfb3ba8eed56

## Pre-Merge Checklist
